dnadlinger/oitg
# rpe analyse

- results: Avoid ARTIQ dependency by parsing expid as JSON
- results: Allow using HDF5 loading code with known file path
- circuits: Gate sequence generation, execution and analysis
- results: RST-ify doc comments and add to Sphinx build
- results: Avoid deprecated h5py .value accessor
- results: flake8
- results: YAPF [nfc]
- CircleCI: Add initial configuration for CI/doc builds
- docs/index: This sentence now verb
- circuits/runner: Make dataset_prefix a run_sequences() argument
- circuits/protocols/gst: Add some (obvious) return types
- dipole_bsb_car_rsb: fixed imports
- fitting\sinusoid: Robust general sinusoid with dead time This fitting routine is intended to replace the specialised sinusoid fitting routines.
- fitting\sinusoid: flake8
- fitting\sinusoid: fixed dead-time bug for data not starting at t=0.
- fitting\dipole_bsb_car_rsb: improved frequency initialisation & docs
- docs: Fix references to ndscan in Sphinx config
- Robust decaying sinusoid fitting
- fitting/sinusoid: expanded docstring and added to top-level module
- sinusoid fitting test case (#11)
- fitting: Import decaying_sinusoid into package
- Improved Sinc^2 fitting routine (#12)
- fittig: Fix a few comment typos
- README: Link to github.io doc build
- Remove no-op code from package __init__
- CircleCI: Run test/fitting unit tests
- fitting: Fail on undefined constants/initial values
- tightened fourier limit to pass fitting tests
- threshold: find optimal readout threshold and bin time (#13)
- threshold: Add to API doc build
- Update build container with new YAPF/Flake8 versions
- YAPF/Flake8-ify recent modules; add to whitelist
- CI: Fix up YAPF whitelist
- test_sinusoid: Relax randomised tests to 4σ
- Rabi-oscillation detuning scan with a floated pulse area.  (#21)
- fitting\detuned_square_pulse: flake8 (nfc)
- fitting\sinusoid: flake8 (nfc)
- fitting\decaying_sinusoid: flake8 (nfc)
- fitting\FitBase: flake8 (nfc)
- fitting\dipole_bsb_car_rsb: flake8 (nfc)
- test\fitting\test_detuned_square_pulse: flake8 (nfc)
- fitting/test_detuned_square_pulse: yapf (nfc)
- circuits/protocols/gst/specs: Fix docstring copypasta [nfc]
- circuits/protocols/gst/specs: Add make_2q_xy_pi_2_wobble_spec()
- circuits.protocols.gst: Add optimised X, Z, wobble gate set spec
- circuits: Add (spin-echo'd) wobble gate two-qubit Clifford group enumeration
- add roi finding tool for camera image analysis
- circuits: changes by DPN for blade rbm?
- imaging: fix init file
- setup.py: Add statsmodel dependency
- circuits.protocols.process_tomo: Allow specifying SPAM projectors in prepare_data()
- errorbars: Fix factor of two in confidence interval size
- circuits: Fix CI build (flake8)
- results: Remove duplicate int() conversion [nfc]
- results: Support all find_results() parameters in load_result()
- paths: YAPF-ify, add to CI whitelist
- docs: Add oitg.paths API docs to the build
- results: Flip filter arguments order to match hierarchy; document all of them
- paths: Style/documentation; shared_{dir -> area}_path()
- paths: Add type annotations
- results: Add type annotations
- results: Fix/clarify comments
- conda: Add statsmodels as build dependency
- oitg.fitting: Add detuned_square_pulse to __init__
- FitBase: Don't set scale for constant parameters The scale is not needed for constant parameters. Further, the check `constant_parameters[name] != 0` is incompatible with the kludge in `dipole_bsb_car_rsb` where a numpy array is passed as a constant.
- yaml: update to pyton 3.8
- results: Slightly clean up find_results() in preparation for auto-search [nfc]
- CircleCI: Bump container to version with Python 3.8
- Incomplete rid_index/oitg_fetch draft assuming ordered RIDs
- rid_index: Rewrite to store each RID
- CircleCI: Add RID lookup modules to flake8/YAPF whitelist
- results: Fix magic string matching regex
- paths: Respect OITG_SHARED_AREA in shared_area_path
- setup.py: Add frontend scripts
- Add oitg.frontend __init__ module
- threshold: rework to include de-shelving effects (breaking change) The commit accounts for deviation from poissonian statistics due to de-shelving of the dark state. The treatement follows chapeter 3 of the thesis: Alice Burrell, 2010 closes #26
- threshold: Unbreak CI (YAPF-ify) [nfc]
- circuits.protocol.rbm: Fix 2Q circuit generation example
- test_threshold: YAPF-ify to unbreak CI
- results: Better error message when shared area not mounted
- Implement RBM on the qutrit-like symmetric subspace of two qubit states
- circuits: Add qutrit subspace Clifford group decomposition for -XX gate
- circuits.clifford_sym_qutrit: Some more round numbers for mxx decomposition
- circuits.clifford_sym_qutrit: Add Z⊗Z wobble gate decomposition
- clifford_sym_qutrit: Document data table approach [nfc]
- circuits.protocols.process_tomo.analyse: fixup argument name
- circuits.protocols.process_tomo.generate: Split up sequence generation function
- results: Fix for Python 3.10
- process_tomo.generate: Fix flake8 whitespace nits breaking CI [nfc]
- results: read archive in addition to datasets
- results: Add optional local_path (to automatically keep and use a local copy)
- circuits.protocols.rpe: Factor out core phase estimation loop
